### ISSUE TITLE
Using SAM image encoder with different image size

### ISSUE THREAD

**Issue by aminpdik** [facebookresearch/segment-anything#374]
Thanks for the great work.

@likethesky @Celebio @colesbury @pdollar 
@minqi 

I am gonna try SAM image encoder with image size 400*400. But the model weights are for size 1024*1024 and I get mismatch error. Is there any way to use SAM with different image sizes. 

I think It would be great for the vision community to be able to use SAM encoder for feature extraction.

**Comment by Scott-che** [comment]
same issue

**Comment by chaoer** [comment]
maybe you can see:
[def resize_pos_embed(](https://github.com/huggingface/pytorch-image-models/blob/2d0dbd17e388953ab81a5c56f80074eff962ea6b/timm/models/vision_transformer.py#L689)

**Comment by Scott-che** [comment]
Thanks your help

**Comment by nazimii** [comment]
@chaoer  can it work without finetune the model at other image size?

**Comment by aminpdik** [comment]
> > @chaoer can it work without finetune the model at other image size?
> 
> yeap. I successfully finetune SAM (ViT-H) with image size 256 or any size else.
> 
> But you should modify this code slightly to adapt to the weights of `rel_pos_w` and `rel_pos_h`

Great Work @chaoer 

Is it possible to provide the modified code?

**Comment by chaoer** [comment]
> > > @chaoer can it work without finetune the model at other image size?
> > 
> > 
> > yeap. I successfully finetune SAM (ViT-H) with image size 256 or any size else.
> > But you should modify this code slightly to adapt to the weights of `rel_pos_w` and `rel_pos_h`
> 
> Great Work @chaoer
> 
> Is it possible to provide the modified code?
@aminpdik 
sorry, its not inconvenient to share the whole code for some reasons.

actually, you can directly using the code [here](https://github.com/huggingface/pytorch-image-models/blob/c241081251323dfc5e8dc799d49740c48cc9096f/timm/layers/pos_embed.py#L17)

but you should remember that:
1. we do not have prefix tokens, so set param `num_prefix_tokens` to `0`
2. for position embedding tensor `rel_pos_w `and `rel_pos_h`, you should add dim before invoke this func, and remove it after interpolation. (namely, `posemb = posemb.unsqueeze(0) ,   xxxx,  posemb = posemb.squeeze(0) `)

**Comment by alessandromaranelli** [comment]
Anyone solved this issue and has some reference code?

**Comment by benam2** [comment]
Hi all, Any update on this?
Does that make sense to use Embedding of the SAM model for the retrieval tasks?
Thanks~
